### PR TITLE
txn: add configurations and enable pessimistic-txn by default

### PR DESCRIPTION
**Pull request by youjiali1995**
Signed-off-by: youjiali1995 <zlwgx1023@gmail.com>

## What have you changed? (mandatory)
Make some hard code configuration to be configurable and enable pessimistic-txn by default.

## What are the type of the changes? (mandatory)
- Improvement (change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)
unit tests

## Refer to a related PR or issue link (optional)
https://github.com/tikv/tikv/pull/4698

### CODE REVIEW

**Comment by youjiali1995** [comment]
@zhangjinpeng1987 @AndreMouche @coocood PTAL

File: src/storage/lock_manager/waiter_manager.rs [hunk 1]
```diff
@@ -362,47 +367,34 @@ impl FutureRunnable<Task> for WaiterManager {
...
-    let mut s = DefaultHasher::new();
-    key.hash(&mut s);
-    s.finish()
-}
-
-pub fn gen_key_hashes(keys: &[Key]) -> Vec<u64> {
-    keys.iter().map(|key| gen_key_hash(key)).collect()
+fn wake_up_waiter(waiter: Waiter, commit_ts: u64) {
+    // Maybe we can store the latest commit_ts in TiKV, and use
+    // it as `conflict_start_ts` when waker's `commit_ts` is smaller
```
Comment (coocood) [inline src/storage/lock_manager/waiter_manager.rs]: as `conflict_commit_ts`

File: src/storage/lock_manager/waiter_manager.rs [hunk 2]
```diff
@@ -177,17 +176,22 @@ impl Scheduler {
...
     ) {
-        self.notify_scheduler(Task::WaitFor {
-            start_ts,
-            cb,
-            pr,
-            lock,
-            is_first_lock,
-            timeout,
-        });
+        let timeout = calculate_wait_for_lock_timeout(lock.ts, ttl);
```
Comment (coocood) [inline src/storage/lock_manager/waiter_manager.rs]: Why use `ts` and `ttl` to calculate timeout?

Comment (youjiali1995) [reply src/storage/lock_manager/waiter_manager.rs]: Now the timeout is `min(max_wait_for_lock_timeout, now - (lock's start_ts + lock's ttl))`. If the lock has expired, txns won't wait for it.

Comment (coocood) [reply src/storage/lock_manager/waiter_manager.rs]: We can afford to wait for an expired lock for simpler code.

File: src/storage/lock_manager/util.rs [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+pub const PHYSICAL_SHIFT_BITS: usize = 18;
+
+// In milliseconds
+pub fn extract_physical_timestamp(ts: u64) -> u64 {
+    ts >> PHYSICAL_SHIFT_BITS
+}
+
+pub fn calculate_wait_for_lock_timeout(start_ts: u64, ttl: u64) -> u64 {
+    let limit = Duration::from_millis(extract_physical_timestamp(start_ts) + ttl);
+    let now = SystemTime::now()
```
Comment (AndreMouche) [inline src/storage/lock_manager/util.rs]: what if the current machine's logical time not correct or has a big difference with `pd`?

File: src/storage/lock_manager/config.rs [hunk 4]
```diff
@@ -2,15 +2,54 @@
...
 use std::error::Error;
 
-#[derive(Clone, Serialize, Deserialize, Default, PartialEq, Debug)]
+#[derive(Clone, Serialize, Deserialize, PartialEq, Debug)]
 #[serde(default)]
 #[serde(rename_all = "kebab-case")]
 pub struct Config {
     pub enabled: bool,
+    pub wait_for_lock_timeout: u64,
+    pub wake_up_dealy_duration: u64,
```
Comment (coocood) [inline src/storage/lock_manager/config.rs]: delay

File: src/storage/lock_manager/util.rs (hunk 3 above)
Comment (youjiali1995) [reply src/storage/lock_manager/util.rs]: It has been removed, and it doesn't matter.

**Comment by coocood** [comment]
LGTM

**Comment by youjiali1995** [comment]
ping @zhangjinpeng1987

**Pull request merged**